File: frontend/src/types/booking.ts
/**
 * Booking / trips TypeScript types (mirror backend BookingPublic).
 */

export type BookingStatus = "pending" | "confirmed" | "cancelled" | "completed";

export interface BookingListingSummary {
  id: number;
  title: string;
  city: string;
  country: string;
  image_url: string | null;
}

export interface Booking {
  id: number;
  listing_id: number;
  guest_id: number;
  check_in: string;
  check_out: string;
  guests: number;
  status: BookingStatus;
  nightly_rate_cents: number;
  cleaning_fee_cents: number;
  service_fee_cents: number;
  total_cents: number;
  created_at: string;
  listing: BookingListingSummary | null;
  has_review?: boolean;
  can_review?: boolean;
}

export interface TripsResponse {
  upcoming: Booking[];
  past: Booking[];
  cancelled: Booking[];
}

export interface CreateBookingPayload {
  listing_id: number;
  check_in: string;
  check_out: string;
  guests: number;
}

File: backend/app/schemas/booking.py
"""
Booking API schemas.

Why this file exists:
  Request/response shapes for create, checkout, cancel, and My Trips.
  Totals are always server-computed; clients never send money amounts.
"""

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import BookingStatus


class BookingCreate(BaseModel):
  listing_id: int = Field(gt=0)
  check_in: date
  check_out: date
  guests: int = Field(ge=1)


class BookingListingSummary(BaseModel):
  """Listing snapshot embedded on trip / checkout cards."""

  model_config = ConfigDict(from_attributes=True)

  id: int
  title: str
  city: str
  country: str
  image_url: str | None = None


class BookingPublic(BaseModel):
  model_config = ConfigDict(from_attributes=True)

  id: int
  listing_id: int
  guest_id: int
  check_in: date
  check_out: date
  guests: int
  status: BookingStatus
  nightly_rate_cents: int
  cleaning_fee_cents: int
  service_fee_cents: int
  total_cents: int
  created_at: datetime
  listing: BookingListingSummary | None = None
  has_review: bool = False
  can_review: bool = False


class TripsResponse(BaseModel):
  upcoming: list[BookingPublic]
  past: list[BookingPublic]
  cancelled: list[BookingPublic]
